component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/constant/Constants.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.extension.identity.emailotp.common.constant;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Email OTP service constants.
+ */
+public class Constants {
+
+    public static final String ALGORITHM_NAME = "SHA1PRNG";
+    public static final String ALGORITHM_HMAC = "HmacSHA1";
+    public static final String ALGORITHM_HMAC_SHA = "HMAC-SHA-1";
+    public static final String SESSION_TYPE_OTP = "EMAIL_OTP";
+    public static final String NOTIFICATION_TYPE_EMAIL_OTP = "EmailOTP";
+
+    public static final int NUMBER_BASE = 2;
+    public static final int DEFAULT_OTP_LENGTH = 6;
+
+    public static final int DEFAULT_EMAIL_OTP_EXPIRY_TIME = 120000;
+    public static final int DEFAULT_EMAIL_RESEND_THROTTLE_INTERVAL = 30000;
+
+    public static final String EMAIL_OTP_IDENTITY_EVENT_MODULE_NAME = "emailOtp";
+    public static final String EMAIL_OTP_ENABLED = "emailOtp.enabled";
+    public static final String EMAIL_OTP_LENGTH = "emailOtp.tokenLength";
+    public static final String EMAIL_OTP_VALIDITY_PERIOD = "emailOtp.tokenValidityPeriod";
+    public static final String EMAIL_OTP_ALPHA_NUMERIC_OTP = "emailOtp.isEnableAlphanumericToken";
+    public static final String EMAIL_OTP_TRIGGER_OTP_NOTIFICATION = "emailOtp.triggerNotification";
+    public static final String EMAIL_OTP_RENEWAL_INTERVAL = "emailOtp.tokenRenewalInterval";
+    public static final String EMAIL_OTP_RESEND_THROTTLE_INTERVAL = "emailOtp.resendThrottleInterval";
+    public static final String EMAIL_OTP_SHOW_FAILURE_REASON = "emailOtp.showValidationFailureReason";
+
+    /**
+     * EMAIL OTP service error codes.
+     */
+    public enum ErrorMessage {
+
+        // Client error codes.
+        CLIENT_BAD_REQUEST("EMAIL-60001", "Bad request.", "Bad request : %s."),
+        CLIENT_EMPTY_USER_ID("EMAIL-60002", "Provided user ID is empty.", "Provided user ID is empty."),
+        CLIENT_INVALID_USER_ID("EMAIL-60003", "Invalid user Id.", "Provided user ID is invalid : %s."),
+        CLIENT_BLANK_EMAIL_ADDRESS("EMAIL-60004", "Invalid email address.",
+                "No valid email address for the user : %s."),
+        CLIENT_EXPIRED_OTP("EMAIL-60005", "Expired OTP.",
+                "Expired OTP."),
```

Comment:
This line can be moved to the above. Please check the formatting in ErrorMessages. Don't break lines unless the character length is greater than 120

Reply:
fixed in 0e37ffc